x/concentrated-liquidity/swaps.go

```diff
@@ -187,7 +187,7 @@ func (k *Keeper) swapInAmtGivenOut(
 	swapFee sdk.Dec,
 	priceLimit sdk.Dec,
 ) (calcTokenIn, calcTokenOut sdk.Coin, currentTick int64, liquidity, sqrtPrice sdk.Dec, err error) {
-	tokenIn, tokenOut, newCurrentTick, newLiquidity, newSqrtPrice, err := k.computeInAmtGivenOut(ctx, desiredTokenOut, tokenInDenom, swapFee, priceLimit, pool.GetId())
+	writeCtx, tokenIn, tokenOut, newCurrentTick, newLiquidity, newSqrtPrice, err := k.calcInAmtGivenOut(ctx, desiredTokenOut, tokenInDenom, swapFee, priceLimit, pool.GetId())
```

Comment:
we don't need to call CalcInAmtGivenOut and then call writeCtx, instead, here we pass in computeInAmtGivenOut directlyt with the context we are using, thus mutating state anyways